lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_generate_apk_from_aab.rb

```diff
@@ -0,0 +1,132 @@
+          raise
+        end
+
+        # If no AAB param was provided, attempt to get it from the lane context
+        # First GRADLE_ALL_AAB_OUTPUT_PATHS if only one
+        # Second GRADLE_AAB_OUTPUT_PATH if it is set
+        # Else use the specified parameter value
+        if params[:aab_file_path].nil?
+          all_aab_paths = Actions.lane_context[SharedValues::GRADLE_ALL_AAB_OUTPUT_PATHS] || []
+          aab_file_path = if all_aab_paths.count == 1
+                            all_aab_paths.first
+                          else
+                            Actions.lane_context[SharedValues::GRADLE_AAB_OUTPUT_PATH]
+                          end
+        else
+          aab_file_path = params[:aab_file_path]
+        end
+
+        # If no AAB file path was found, raise an error
+        if aab_file_path.nil?
+          UI.user_error!('No AAB file path was specified and none was found in the lane context. Please specify the `aab_file_path` parameter or ensure that the relevant build action has been run prior to this action.')
+          raise
+        end
+
+        apk_output_file_path = params[:apk_output_file_path]
+        keystore_path = params[:keystore_path]
+        keystore_password = params[:keystore_password]
+        keystore_key_alias = params[:keystore_key_alias]
+        signing_key_password = params[:signing_key_password]
+
+        sh(generate_command(aab_file_path, apk_output_file_path, keystore_path, keystore_password, keystore_key_alias, signing_key_password))
+      end
+
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.description
+        'Generates an APK from the specified AAB'
+      end
+
+      def self.details
+        'Generates an APK from the specified AAB'
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(
+            key: :aab_file_path,
+            env_name: 'ANDROID_AAB_FILE_PATH',
+            description: 'The path to the AAB file. If not speicified, the action will attempt to read from the lane context using the `SharedValues::GRADLE_ALL_AAB_OUTPUT_PATHS` and `SharedValues::GRADLE_AAB_OUTPUT_PATH` keys',
+            type: String,
+            optional: true,
+            default_value: nil,
+            verify_block: proc { |p| UI.user_error!("AAB path `#{p}` is not a valid file path.") unless File.file?(p) }
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :apk_output_file_path,
+            env_name: 'ANDROID_APK_OUTPUT_PATH',
+            description: 'The output path where the APK file will be generated. The directory will be created if it does not yet exist',
+            type: String,
+            optional: false,
+            default_value: nil
```

Comment:
1️⃣  If that `ConfigItem` is not optional (`optional: false`) because we want to make sure the user provides an output path, then it doesn't make sense to have a `default_value` there.

```suggestion
            optional: false
```

Alternatively, we could consider that it could be valid to not pass any value for that parameter and in such case have the action infer its value as being the same path as `aab_file_path` but with `.aab` extension replaced by `.apk`. (suggestion 2️⃣ ) below)

Reply:
> Alternatively, we could consider that it could be valid to not pass any value for that parameter and in such case have the action infer its value as being the same path as `aab_file_path` but with `.aab` extension replaced by `.apk`.

Now implemented in [0e801e1](https://github.com/wordpress-mobile/release-toolkit/pull/467/commits/0e801e1be5b514e1b8c8c7717881991e916021a0#diff-89bdefd7112342eae6bc39216c324717a41f958babd4374b1bc30e3a47a7c0b1R13) via this line: https://github.com/wordpress-mobile/release-toolkit/blob/1d566d1e235dad84f79783a82e70378e6cbb865a/lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_generate_apk_from_aab.rb#L13